client/my-sites/themes/theme-options.js

```diff
@@ -74,6 +77,34 @@ const activateOnJetpack = {
 	)
 };
 
+const deleteConfirmation = ( themeId, siteId ) => {
+	return ( dispatch, getState ) => {
+		const { name: themeName } = getTheme( getState(), siteId, themeId );
+		const siteSlug = getSiteSlug( getState(), siteId );
+		accept(
+			i18n.translate(
+				'Are you sure you want to delete %(themeName)s from %(siteSlug)s?',
+				{ args: { themeName, siteSlug }, context: 'Themes: theme delete confirmation dialog' }
+			),
+			( accepted ) => {
+				accepted && dispatch( deleteThemeAction( themeId, siteId ) );
+			},
+			i18n.translate(
+				'Delete %(themeName)s',
+				{ args: { themeName }, context: 'Themes: theme delete dialog confirm button' }
+			),
+			i18n.translate( 'Back', { context: 'Theme: theme delete dialog back button' } )
+		);
+	};
+};
+
+const deleteTheme = {
```

Comment:
Hmm, too bad we can't just call this `delete` to stick with the naming convention here since that's a reserved word...